Cpp_test/lib/ipc-pipe-send.cpp

```diff
@@ -0,0 +1,70 @@
+#include "ipc-pipe-send.h"
+
+IPCPipeSend::~IPCPipeSend()
+{
+	close(this->pd);
+	unlink(this->info.method_name);
+}
+
+void IPCPipeSend::init()
+{
+	this->file_handler.setup_file(info.file_name, FileMode::READ);
+	// remove old pipe (if any)
+	remove(this->info.method_name);
+	// create a fifo
+	if (mkfifo(this->info.method_name, 0660) != 0){
+        throw std::runtime_error(std::string("ERROR: create pipe with send name ") + this->info.method_name );
+    }
+		
+	// open the pipe
+	this->pd = open(this->info.method_name, O_RDWR | O_NONBLOCK);
```

Comment:
Is there a reason of O_RDWR and not write only?

Same as the receiver, if you use the O_NONBLOCK flag you should take care about race issues.

Reply:
For instance, the user must launch the sender and receiver at the same time to be sure that every data is passed.

Reply:
I will set sleep time for receiver.